internal/service/projectipaccesslist/resource_project_ip_access_list.go

```diff
@@ -200,50 +200,15 @@ func (r *projectIPAccessListRS) Create(ctx context.Context, req resource.CreateR
 		return
 	}
 
-	projectIPAccessListNewModel := newTFProjectIPAccessListModel(projectIPAccessListModel, entry)
+	projectIPAccessListNewModel := NewTfProjectIPAccessListModel(projectIPAccessListModel, entry)
 	resp.Diagnostics.Append(resp.State.Set(ctx, &projectIPAccessListNewModel)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 }
 
-func newTFProjectIPAccessListModel(projectIPAccessListModel *tfProjectIPAccessListModel, projectIPAccessList *admin.NetworkPermissionEntry) *tfProjectIPAccessListModel {
-	entry := projectIPAccessList.GetIpAddress()
-	if projectIPAccessList.GetCidrBlock() != "" {
-		entry = projectIPAccessList.GetCidrBlock()
-	} else if projectIPAccessList.GetAwsSecurityGroup() != "" {
-		entry = projectIPAccessList.GetAwsSecurityGroup()
-	}
-
-	id := conversion.EncodeStateID(map[string]string{
-		"entry":      entry,
-		"project_id": projectIPAccessList.GetGroupId(),
-	})
-
-	return &tfProjectIPAccessListModel{
-		ID:               types.StringValue(id),
-		ProjectID:        types.StringValue(projectIPAccessList.GetGroupId()),
-		CIDRBlock:        types.StringValue(projectIPAccessList.GetCidrBlock()),
-		IPAddress:        types.StringValue(projectIPAccessList.GetIpAddress()),
-		AWSSecurityGroup: types.StringValue(projectIPAccessList.GetAwsSecurityGroup()),
-		Comment:          types.StringValue(projectIPAccessList.GetComment()),
-		Timeouts:         projectIPAccessListModel.Timeouts,
-	}
-}
-
-func newMongoDBProjectIPAccessList(projectIPAccessListModel *tfProjectIPAccessListModel) *[]admin.NetworkPermissionEntry {
```

Comment:
is it ok to assume tfProjectIPAccessListModel is never nil?

Reply:
yes, it will never be null. Also, it's attributes `AWSSecurityGroup`,  `CIDRBlock` and `IPAddress` are also checked